src/services/findAllReferences.ts

```diff
@@ -1407,49 +1418,38 @@ namespace ts.FindAllReferences.Core {
              * will be included correctly.
              */
             const shorthandValueSymbol = checker.getShorthandAssignmentValueSymbol(location.parent);
-            if (shorthandValueSymbol) {
-                result.push(shorthandValueSymbol);
-            }
+            const res1 = shorthandValueSymbol && cbSymbol(shorthandValueSymbol);
+            if (res1) return res1;
         }
 
+        const res = fromRoot(symbol);
+        if (res) return res;
+
         // If the symbol.valueDeclaration is a property parameter declaration,
         // we should include both parameter declaration symbol and property declaration symbol
         // Parameter Declaration symbol is only visible within function scope, so the symbol is stored in constructor.locals.
         // Property Declaration symbol is a member of the class, so the symbol is stored in its class Declaration.symbol.members
-        addRange(result, getParameterPropertySymbols(symbol, checker));
+        if (symbol.valueDeclaration && isParameter(symbol.valueDeclaration) && isParameterPropertyDeclaration(symbol.valueDeclaration)) {
+            const paramProps = checker.getSymbolsOfParameterPropertyDeclaration(symbol.valueDeclaration, symbol.name);
+            const res = paramProps && cbParameterProperties(paramProps);
+            if (res) return res;
+        }
 
         // If this is symbol of binding element without propertyName declaration in Object binding pattern
         // Include the property in the search
         const bindingElementPropertySymbol = getPropertySymbolOfObjectBindingPatternWithoutPropertyName(symbol, checker);
-        if (bindingElementPropertySymbol) {
-            result.push(bindingElementPropertySymbol);
-            addRootSymbols(bindingElementPropertySymbol);
-        }
-
-        addRootSymbols(symbol);
-
-        return result;
-
-        function addRootSymbols(sym: Symbol): void {
-            // If this is a union property, add all the symbols from all its source symbols in all unioned types.
-            // If the symbol is an instantiation from a another symbol (e.g. widened symbol) , add the root the list
-            for (const rootSymbol of checker.getRootSymbols(sym)) {

```

Comment:
Can you please move this comment may be at ```fromRoot(symbol)``` so we know we are iterating over symbols for unioned types? Thanks

Reply:
Moved comment to fromRoot and elaborated (since behavior is different in the two cases)